staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -649,6 +648,14 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		if err != nil {
 			return err
 		}
+		if r.timeout > 0 {
+			if r.ctx == nil {
+				r.ctx = context.Background()
+			}
+			var cancelFn context.CancelFunc
+			r.ctx, cancelFn = context.WithTimeout(r.ctx, r.timeout)
+			defer cancelFn()
```

Comment:
I think this works, but it will wait until all the retries are done.  I'm fine with it since it eventually cleans up.  I'll give some time if people really want to argue that it all ends up as an anonymous function to clean up sooner in exceptional cases.  I can't see it mattering much.